### ISSUE TITLE
TypeCast won't check objects accordingly in Vscode - TypeScript 4.6.3 VsCode 1.66.2

### ISSUE THREAD

**Issue by SimonPiscitelli** [microsoft/TypeScript#49320]
# Bug Report

### 🔎 Search Terms

vscode
type cast

### 🕗 Version & Regression Information

- Visual Studio Code v 1.66.2
- (vscode)TypeScript version 4.6.3 
- This changed between versions 4.2 and 4.6
- I was unable to test this on prior versions because a lot of work to do
- 
### 💻 Code

```ts
interface TestInterface {
    foo: string;
    bar: string;
}

// good
let object1: TestInterface = {
    foo: "",
    bar: "",
};

// linter error: Object literal may only specify known properties, and 'baz' does not exist in type 'TestInterface'
// good
let object2: TestInterface = {
    foo: "",
    bar: "",
    baz: "",
};

// no linter error
// bad
let object3 = <TestInterface>{
    foo: "",
    bar: "",
    baz: "",
};

// no linter error
// bad
let object4 = {
    foo: "",
    bar: "",
    baz: "",
} as TestInterface
```

### 🙁 Actual behavior

TypeCasted objects doesn't trigger linter error

### 🙂 Expected behavior

Linter should have flagged the same error for object3 than for object2

**Comment by MartinJohns** [comment]
This is working as intended. Using a type assertion you tell the compiler to trust you and ignore the missing properties and to turn off excess-property checks.
